### PR TITLE
Added description for authentication types

### PR DESCRIPTION
**Pull request by siddhartha97**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Ensures all external APIs in `staging/src/k8s.io/api/authentication/v*/types.go` have descriptions.
#### Which issue(s) this PR fixes:

Ref #99675

CC: @nikhita 
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/api/authentication/v1/types.go [hunk 1]
```diff
@@ -130,10 +132,13 @@ func (t ExtraValue) String() string {
...
 type TokenRequest struct {
 	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
 	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
-
+	// Spec holds information about the request being evaluated
 	Spec TokenRequestSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	// Status is filled in by the server and indicates whether the request can be authenticated.
```
[the remark this answers is not in the document]
Comment (liggitt) [reply staging/src/k8s.io/api/authentication/v1/types.go:141]: also change in the v1beta1 doc

[the remark this answers is not in the document]
Comment (siddhartha97) [reply staging/src/k8s.io/api/authentication/v1/types.go]: @liggitt while trying to merge new change I messed up the git history :man_facepalming: , so I pushed the code as a new PR #101687 

Would really appreciate if you could take a look at it! Really sorry for the inconvenience

**Pull request closed without merging**